### PR TITLE
SONAR-7336 ability to change issue type

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/issue/SetTypeAction.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return true;
+  }
+
+  @Override
+  public boolean execute(Map<String, Object> properties, Context context) {
+    String type = newValue(properties);
+    return issueUpdater.setType((DefaultIssue) context.issue(), RuleType.valueOf(type), context.issueChangeContext());
+  }
+
+  private String newValue(Map<String, Object> properties) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/SetTypeAction.java]: MAJOR Make "newValue" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-web/src/main/js/components/issue/views/set-type-form-view.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  },
+
+  selectInitialOption () {
+    return this.makeActive(this.getOptions().filter(`[data-value="${this.getType()}"]`));
+  },
+
+  selectOption (e) {
+    const issueType = $(e.currentTarget).data('value');
+    this.submit(issueType);
+    return ActionOptionsView.prototype.selectOption.apply(this, arguments);
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/issue/views/set-type-form-view.js]: MAJOR Use the rest syntax to declare this function's arguments. [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3AS3513)

File: server/sonar-web/src/main/js/components/issue/views/set-type-form-view.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  },
+
+  submit (issueType) {
+    return this.model.setType(issueType);
+  },
+
+  serializeData () {
+    return _.extend(ActionOptionsView.prototype.serializeData.apply(this, arguments), {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/issue/views/set-type-form-view.js]: MAJOR Use the rest syntax to declare this function's arguments. [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3AS3513)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by teryk** [comment]
LGTM, reorganize commits to not have a rebase master commit

**Comment by simonbrandhof** [comment]
of course, I usually don't do such commit refactoring on my PR.

**Pull request closed without merging**